webdictionary/src/main/java/com/eis0/kademlianetwork/KademliaMessage.java

```diff
@@ -0,0 +1,105 @@
+package com.eis0.kademlianetwork;
+
+import com.eis.smslibrary.SMSMessage;
+import com.eis.smslibrary.SMSPeer;
+import com.eis0.kademlia.KademliaId;
+
+/**
+ * This class is used to create custom messages to send or receive inside the Kademlia Network, the
+ * fields are in common for all the type of requests
+ */
+public class KademliaMessage {
```

Comment:
A much better solution, but it has the same problems of hardwiring as before. Anyway, you can easily solve them just defining some constants. I think having public fields is fine as long as this class remains private inside your package

Reply:
Yeah, the positions of the fields are still hardwired, I could change them with some constants, but I'm still not sure if it's worth it, considering that that's the only place where they are used, the method itself is quite small and self-explanatory, and inside the method description there is a list of the position of the parameters inside the message; it takes nothing to change them, but with this new structure I honestly don't feel the need to define those constants anymore.

Reply:
Count that like this it's not much expandable, it's your choice if it's worth it. The important is that the logic of syntax is well contained in just one point of access.
I would suggest adding a description of the syntax.